### ISSUE TITLE
Can we have a name field for I/O backends (analagous to plotting)

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#1865]
I find it useful to be able to ask a backend what it's name is (so we can check what a test is meant to do, for instance). We can for plotting (hopefully this remains after #1382) but not for I/O and I think we should add it for I/O.

```
>>> from sherpa import plot
>>> plot.backend.name
'pylab'
>>> plot.backend
<module 'sherpa.plot.pylab_backend' from '/soft/ciao-4.15/lib/python3.9/site-packages/sherpa/plot/pylab_backend.py'>
>>> from sherpa.astro import io
>>> io.backend.name
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'sherpa.astro.io.crates_backend' has no attribute 'name'
>>> io.backend
<module 'sherpa.astro.io.crates_backend' from '/soft/ciao-4.15/lib/python3.9/site-packages/sherpa/astro/io/crates_backend.py'>
```

**Comment by hamogu** [comment]
I think we should actually switch both io and plotting backends to use entrypoints once we drop Python 3.9. If we do that,  we'll have to do a slight rework of the api anyway,  so it might be good to not touch io backends in this release,  in order to avoid adding something else that we need to worry to keep without breaking api.

**Comment by hamogu** [comment]
And while we are at it, can we have a `TemporaryIOBackend` context manager al la #1382?

**Comment by DougBurke** [comment]
Fixed by #2092 (the name field, at least).
